### DISCUSSION TITLE
Check Token and Refresh

### DISCUSSION THREAD

**Discussion by damidu10** [alchemyguy/YoutubeLaravelApi#38] [Q&A]
Hi all,
this library is very useful. I'm using very well.
I need now to check if the authtoken is expired and if it is, refresh it.

Do you have any suggestion?

regards,
Davide

**Comment by alchemyguy** [comment]
@damidu10 

when using the setToken in the package it checks whether its expired or not. if its expired it generates a new token and sets it.
you can refer [this part of the code](https://github.com/alchemyguy/YoutubeLaravelApi/blob/2df1dee7e4f479b8c2d21cc21d6f594df078feec/src/Auth/AuthService.php#L102-L107)
